.github/workflows/lint.yml

```diff
@@ -55,11 +57,25 @@ jobs:
   test:
     timeout-minutes: 5
     runs-on: ubuntu-latest
-    container:
-      image: docker://rust:1.60
     steps:
+      - name: Install rust
+        run: |
+          rm -rf $HOME/.cargo
+          curl --proto =https --tlsv1.3 -fLsSv https://sh.rustup.rs | sh -s -- -y --default-toolchain "${RUST_VERSION}"
+          source $HOME/.cargo/env
+          echo "PATH=$PATH" >> $GITHUB_ENV
+          cargo version
+      # Setup a cluster
+      - run: curl --proto =https --tlsv1.3 -fLsSv https://raw.githubusercontent.com/k3d-io/k3d/${K3D_VERSION}/install.sh | bash
+      - run: k3d --version
+      - run: k3d cluster create --no-lb --k3s-arg '--no-deploy=local-storage,traefik,servicelb,metrics-server@server:*'
+      - run: kubectl version
+      # Install CRDs
+      - run: kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.4.3/config/crd/v1alpha2/gateway.networking.k8s.io_gateways.yaml
```

Comment:
I think it would probably be better to just checkout the whole repo (using actions/checkout) and apply all versions of all of the CRDs.